src/SegmentedMsg.chpl

```diff
@@ -1368,6 +1368,48 @@ module SegmentedMsg {
     var repMsg = "%jt".format(rtn);
     return new MsgTuple(repMsg, MsgType.NORMAL);
   }
+
+  proc segmentedSubstringMsg(cmd: string, payload: string, argSize: int, st: borrowed SymTab): MsgTuple throws {
+    var pn = Reflection.getRoutineName();
+    var repMsg: string;
+
+    var msgArgs = parseMessageArgs(payload, argSize);
+    var objtype = msgArgs.getValueOf("objType");
+    var name = msgArgs.getValueOf("name");
+    
+    // check to make sure symbols defined
+    st.checkTable(name);
+
+    select (objtype) {
+      when ("str") {
+        var strings = getSegString(name, st);
+        var returnOrigins = msgArgs.getValueOf("returnOrigins").toLower(): bool;
+        if returnOrigins {
+          var (off, byt, longEnough) = strings.getFixes(msgArgs.getValueOf("nChars"): int,
+                                                        msgArgs.getValueOf("kind"): Fixes,
+                                                        msgArgs.getValueOf("proper").toLower(): bool,
+                                                        true);
+          var retString = getSegString(off, byt, st);
+          var leName = st.nextName();
+          st.addEntry(leName, new shared SymEntry(longEnough));
+          repMsg = "created " + st.attrib(retString.name) + "+created bytes.size %t".format(retString.nBytes) + "+created " + st.attrib(leName);
+        } else {
+          var (off, byt) = strings.getFixes(msgArgs.getValueOf("nChars"): int,
+                                            msgArgs.getValueOf("kind"): Fixes,
+                                            msgArgs.getValueOf("proper").toLower(): bool,
+                                            false);
+          var retString = getSegString(off, byt, st);
+          repMsg = "created " + st.attrib(retString.name) + "+created bytes.size %t".format(retString.nBytes);
+        }
```

Comment:
I think we simplify things a bit if we wait til the end to do the `if/else` on `returnOrigins`

```Chapel
var (off, byt, longEnough) = strings.getFixes(msgArgs.get("nChars").getIntValue(),
                                              msgArgs.getValueOf("kind"): Fixes,
                                              msgArgs.get("proper").getBoolValue());
var retString = getSegString(off, byt, st);
repMsg = "created " + st.attrib(retString.name) + "+created bytes.size %t".format(retString.nBytes);
if returnOrigins {
  var leName = st.nextName();
  st.addEntry(leName, new shared SymEntry(longEnough));
  repMsg += "+created " + st.attrib(leName);
}
```

This  also allows us to eliminate the 2 versions of `getFixes` based on `returnOrigins` since `getFixesHelper` always returns what we want now